odoo/addons/base/tests/test_ir_mail_server.py

```diff
@@ -455,3 +456,60 @@ def test_mail_server_mail_default_from_filter(self):
             message_from='specific_user@example.com',
             from_filter='example.com',
         )
+
+    @mute_logger('odoo.models.unlink')
+    @patch.dict('odoo.tools.config.options', {'from_filter': 'fake.com', 'smtp_server': 'cli_example.com'})
+    def test_mail_server_ir_mail_server_cli(self):
+        """Check the mail server when the "smtp_authentication" is "cli".
+
+        Should take the configuration from the odoo-bin argument.
+        The "from_filter" of the mail server should overwrite the one set
+        in the CLI arguments.
+        """
+        # should be ignored by the mail server
+        self.env['ir.config_parameter'].sudo().set_param('mail.default.from_filter', 'fake.com')
+
+        self.env['ir.mail_server'].create([{
+            'name': 'Server No From Filter',
+            'smtp_host': 'smtp_host',
+            'smtp_encryption': 'none',
+            'smtp_authentication': 'cli',
+            'from_filter': 'cli_example.com',
+        }])
+
+        IrMailServer = self.env['ir.mail_server']
+
+        with self.mock_smtplib_connection():
+            smtp_session = IrMailServer.connect(smtp_from='test@cli_example.com')
+            message = self._build_email(mail_from='test@cli_example.com')
```

Comment:
Shoudln't we rely on mocked tools config ?

Reply:
indeed, not necessary